hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/constants/FileSystemConfigurations.java

```diff
@@ -127,6 +127,7 @@ public final class FileSystemConfigurations {
   // To have functionality similar to drop1 delete is going to wasb by default for now.
   public static final boolean DEFAULT_FS_AZURE_REDIRECT_RENAME = false;
   public static final boolean DEFAULT_FS_AZURE_REDIRECT_DELETE = true;
+  public static final int DEFAULT_FS_AZURE_MAX_CONSUMER_LAG = 7000;
```

Comment:
How did we arrive at this number ?

Reply:
Since, max number of blobs returned in one go can be 5000. So one two iteration, producer will add 10000 blobs. The consumer will take out one by one.  Wanted to have something between 5000 and 10000. This is a configurable config.
Plus, minor changes in this value will have no difference. This constraint is just to prevent possible OOM.

Reply:
consumerlag is removed. we now have a queue max size = 10000